client/views/view_units.cpp

```diff
@@ -664,12 +664,9 @@ std::vector<unit_view_entry> get_units_view_data()
               client_has_player()
               && nullptr != can_upgrade_unittype(client_player(), unittype);
 
-          // Only units with a home city have upkeep
-          if (punit->homecity != 0) {
-            gold_cost += punit->upkeep[O_GOLD];
-            food_cost += punit->upkeep[O_FOOD];
-            shield_cost += punit->upkeep[O_SHIELD];
-          }
+          gold_cost += punit->upkeep[O_GOLD];
+          food_cost += punit->upkeep[O_FOOD];
+          shield_cost += punit->upkeep[O_SHIELD];
```

Comment:
Shouldn't this depend on `homeless_gold_upkeep`?

Reply:
No need, since the `upkeep` doesn't get populated unless `homeless_gold_upkeep` is enabled. I'm guessing this if-block was either a mistake or premature optimisation. It's better if any upkeep automatically gets added to the view regardless of how the server arrived at the decision to apply it so that it's future-proof against further changes.